### PR TITLE
Eyetoy: Apply first improvement for #4239

### PR DESCRIPTION
**Pull request by Florin9doi**
### Description of Changes
Don't start sending the same frame again, and send NAK instead until a new frame is available

### Rationale behind Changes
Why were these changes made?  What problem does it solve / area does it improve?
#4239

### Suggested Testing Steps
#4239

### CODE REVIEW

**Comment by seta-san** [comment]
fixes
-------
Groove
Play
Play 2
Play 3
Play Hero - needs something long and shiny green to use as a sword
Rhythmic Star
Operation Spy
Astro Zoo - works with some difficulty
Play Sports - works with great difficulty activating menu items

questionable
---------------
Harry Potter and the prisoner of azkaban - menu buttons don't react to your hand but i found that waving something bright green makes them work. i thought that maybe the game came with some sort of wand with a green tip but can't find any evidence of that.


still broken
-------------
Kaiketsu Zorori: Mezase! Itazura King - Limited Edition
antigrav
singstar games
pom pom party - might not be broken the game came with bright green and pink pompoms that might be needed to activate menu items. i tried waving the brightest green thing i have to make it work but no luck... and my wife has a deep seated hatred for pink.

**Comment by seta-san** [comment]
The dumbass things I do to test this emulator

**Comment by seta-san** [comment]
I was wondering. The eyetoy has an led that lets you know if the room is too dark. Is there a way to emulated that?

File: pcsx2/USB/usb-eyetoy/cam-linux.cpp [hunk 1]
```diff
@@ -250,6 +250,25 @@ namespace usb_eyetoy
...
+			CLEAR(fmtd);
+			struct v4l2_frmsizeenum frmsize;
+			CLEAR(frmsize);
+
+			fmtd.index = 0;
+			fmtd.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+			while (xioctl(fd, VIDIOC_ENUM_FMT, &fmtd) >= 0) {
+				frmsize.pixel_format = fmtd.pixelformat;
+				frmsize.index = 0;
+				while (xioctl(fd, VIDIOC_ENUM_FRAMESIZES, &frmsize) >= 0) {
```
Comment (lightningterror) [inline pcsx2/USB/usb-eyetoy/cam-linux.cpp:263]: Bracket should be below, same goes for above.

**Comment by Florin9doi** [comment]
> I was wondering. The eyetoy has an led that lets you know if the room is too dark. Is there a way to emulated that?

Yes, it may be emulated similar to CXBX (https://i.imgur.com/Cjn29ap.png), but it wouldn't be visible in full screen, or creating an overlay using the OSD system (?). I can't implement this, but you should create a new ticket and maybe someone else will implement it based on the following patch: 
```diff
diff --git a/pcsx2/USB/usb-eyetoy/usb-eyetoy-webcam.cpp b/pcsx2/USB/usb-eyetoy/usb-eyetoy-webcam.cpp
index 99c56e610..bbfd12861 100644
--- a/pcsx2/USB/usb-eyetoy/usb-eyetoy-webcam.cpp
+++ b/pcsx2/USB/usb-eyetoy/usb-eyetoy-webcam.cpp
@@ -388,6 +388,16 @@ namespace usb_eyetoy
                                                        Console.WriteLn("EyeToy : configured for unknown format");
                                                }
                                                break;
+                                       case OV519_GPIO_DATA_OUT0:
+                                               {
+                                                       static char led_state = -1;
+                                                       if (led_state != data[0])
+                                                       {
+                                                               led_state = data[0];
+                                                               Console.WriteLn("EyeToy : LED : %d", led_state);
+                                                       }
+                                               }
+                                               break;
                                        case R518_I2C_CTL:
                                                if (data[0] == 1) // Commit I2C write
                                                {
```

**Comment by seta-san** [comment]
will do. also been testing new games. american karaoke revolution games have black screens where the video overlay is supposed to be.

**Review by lightningterror — APPROVED** [review]

**Pull request merged**